### ISSUE TITLE
non-functioning example of service dependency implementations

### ISSUE THREAD

**Issue by sachaw** [arthur-fontaine/diabolo#3]
In the example here: https://diabolo.js.org/docs/implementing-services

The following example:
```ts
import * as DI from 'diabolo'

const otherServiceImpl = DI.lazyCreateServiceImpl<OtherService>(() => {
  const myService = yield * DI.requireService(myService)
  return {
    otherFunction: DI.createFunction(function* () {
      const myService = yield * DI.requireService(myService)
      yield * myService.myFunction()
    })
  }
})
```
Throws an error due to the implementation function not being a generator (no access to yield), making the function an generator make's it incompatible with the expected return signature.

Any solution to this would be much appreciated.
Thanks.

**Comment by arthur-fontaine** [comment]
Hi @sachaw!

Thanks for this issue. Sorry for that. There is indeed a mistake in the code from the documentation. I will fix the documentation asap.

If you need it now, the code should be:

```ts
import * as DI from 'diabolo'

const otherServiceImpl = DI.lazyCreateServiceImpl<OtherService>(() => ({
  otherFunction: DI.createFunction(function* () {
    const myService = yield * DI.requireService(myService)
    yield * myService.myFunction()
  })
}))
```

**Comment by arthur-fontaine** [comment]
Fixed by commit dcbec2e
